### ISSUE TITLE
plan analyze not running automatically

### ISSUE THREAD

**Issue by SlimeDog** [Rsl1122/Plan-PlayerAnalytics#523]
Spigot 1.12.2
Plan 4.1.5

Data are recording to database, but **plan analyze** is not running automatically. Configuration is
```
Analysis:
    AutoRefreshPeriod: 5
```
Analysis ran automatically every 5 minutes in Plan 4.1.5-b3 and previous releases, with this configuration.

No errors in console/log. No obvious issues running with _debug:true_.

**Comment by SlimeDog** [comment]
Two test servers, with only two differences in setup:
WHITE (BLACK backup from 2018-02-12 08:00)
- Essentials 2.0.1-b595
- Plan 4.1.5-b3

BLACK (same backup, updated with two plugins 2018-02-12 08:10)
- Essentials 2.0.1-b596
- Plan 4.1.5

Plan configured for
```
Analysis.AutoRefreshPeriod: 5
```

After running for 8 hours with no players, differences in output:

WHITE
- plan status _Task 229: PeriodicAnalysisTask, Async_ running every 5 minutes
- TPS = 19.89

BLACK
- plan status _PeriodicAnalysisTask, Async_ not running
- TPS = 19.63
- Plan task ids are in the 7000000+

I will update WHITE to Essentials-2.0.1-b596 (I meant to do that before testing overnight - sigh), and retest. That will provide another data point.

**Comment by SlimeDog** [comment]
Test 2

WHITE (BLACK backup from 2018-02-12 08:00, updated with latest Essentials)
- Plan 4.1.5-b3

BLACK (same backup, updated with two plugins 2018-02-12 08:10)
- Plan 4.1.5

Both running
- Essentials 2.0.1-b596
- LuckPerms 4.0.130 running non-verbose mode

WHITE
```
>plan status
[07:39:51 INFO]: » Plan Status
[07:39:51 INFO]:  • Tasks running: 12
[07:39:51 INFO]:  • Processes: 
[07:39:51 INFO]:  • Tasks: 
[07:39:51 INFO]:    Task 216: DBConnectionPingTask SQLite, Async.
[07:39:51 INFO]:    Task 218: ProcessQueueConsumer, Async.
[07:39:51 INFO]:    Task 219: ProcessQueueConsumer, Async.
[07:39:51 INFO]:    Task 220: ProcessQueueConsumer, Async.
[07:39:51 INFO]:    Task 221: ProcessQueueConsumer, Async.
[07:39:51 INFO]:    Task 222: ProcessQueueConsumer, Async.
[07:39:51 INFO]:    Task 223: ProcessQueueConsumer, Async.
[07:39:51 INFO]:    Task 224: ProcessQueueConsumer, Async.
[07:39:51 INFO]:    Task 225: ProcessQueueConsumer, Async.
[07:39:51 INFO]:    Task 226: TPSCountTimer, Sync.
[07:39:51 INFO]:    Task 227: NetworkPageRefreshTask, Async.
[07:39:51 INFO]:    Task 229: PeriodicAnalysisTask, Async.
```

BLACK
```
>plan status
[07:46:00 INFO]: » Plan Status
[07:46:00 INFO]:  • Tasks running: 11
[07:46:00 INFO]:  • Processes: 
[07:46:00 INFO]:    : | Feb 13, 07:43:54 | started   Request:GET /js/demo.js
[07:46:00 INFO]:  • Tasks: 
[07:46:00 INFO]:    Task 216: DBConnectionPingTask SQLite, Async.
[07:46:00 INFO]:    Task 218: ProcessQueueConsumer, Async.
[07:46:00 INFO]:    Task 219: ProcessQueueConsumer, Async.
[07:46:00 INFO]:    Task 220: ProcessQueueConsumer, Async.
[07:46:00 INFO]:    Task 221: ProcessQueueConsumer, Async.
[07:46:00 INFO]:    Task 222: ProcessQueueConsumer, Async.
[07:46:00 INFO]:    Task 223: ProcessQueueConsumer, Async.
[07:46:00 INFO]:    Task 224: ProcessQueueConsumer, Async.
[07:46:00 INFO]:    Task 225: ProcessQueueConsumer, Async.
[07:46:00 INFO]:    Task 226: TPSCountTimer, Sync.
[07:46:00 INFO]:    Task 227: NetworkPageRefreshTask, Async.
```

Note that Task 229: PeriodicAnalysisTask, Async is not running under Plan 4.1.5, but there is a request to js/demo.js (which does not exist, so is running default). I do not see any configuration setting that would drive demo. Configuration files are identical except for
```
% diff plugins/Plan/config.yml /tmp/config.yml 
7c7
<     ServerName: WHITE...
---
>     ServerName: BLACK...
45c45
<     ExternalWebServerAddress: "http://WHITE..."
---
>     ExternalWebServerAddress: "http://BLACK..."
61c61
<         Link: WHITE...:%port%
---
>         Link: BLACK...:%port%
```

**Comment by Rsl1122** [comment]
Try disabling PluginData for Vault if installed.
Another plugin support to disable may be Essentials. 

Essentials uses yml for saving data and is notoriously slow when running analysis and may be the cause here, you might want to check benchmarks on debug page for Source: Essentials etc. If they don't exist, they may be the cause.

**Comment by Rsl1122** [comment]
Also I think luck perms may be using tasks to run something, Plan hardly makes new tasks due to using the 8 Processing queue tasks.

**Comment by SlimeDog** [comment]
When I run /plan analyze manually, I get the following. It does not run automatically.

### Logged Errors
**No Errors logged.**
### Benchmarks
```
10 ms     Analysis: 3rd party
324 ms    Analysis: Data Analysis
205 ms    Analysis: Fetch Phase
1 ms      Analysis: Source ASkyBlock
3 ms      Analysis: Source AdvancedAntiCheat
5 ms      Analysis: Source Economy (Essentials Economy)
5 ms      Analysis: Source Essentials
8 ms      Analysis: Source Factions
545 ms    Analysis: Total
1203 ms   Enable
155 ms    Init Database
12 ms     Initializing locale
1 ms      Processed BukkitConnectionSystem$$Lambda$176/1582822988.
366 ms    Processed InfoSystem$$Lambda$180/1189516313.
1 ms      Processed NetworkSettings$$Lambda$183/1829548004.
1 ms      Processed TPSInsertProcessor.
11 ms     Register Listeners
1 ms      Task Registration
62 ms     WebServer Initialization
```
### Debug Log
```
You're using the latest version.
Using Locale: Default (EN)
Database init..
SQLite database: Opened a new Connection
Started task 216: DBConnectionPingTask SQLite, Async.
Started task 217: DB Clean Task, Async.
sqlite-database connection established.
Initializing Webserver..
No Certificate -> Using Http server for Visualization.
Https Start Failed.
Webserver running on PORT 8804 (http://blackdog.straight8.com:8804)
Started task 218: ProcessQueueConsumer, Async.
Started task 219: ProcessQueueConsumer, Async.
Started task 220: ProcessQueueConsumer, Async.
Started task 221: ProcessQueueConsumer, Async.
Started task 222: ProcessQueueConsumer, Async.
Started task 223: ProcessQueueConsumer, Async.
Started task 224: ProcessQueueConsumer, Async.
Started task 225: ProcessQueueConsumer, Async.
Analysis | Boot analysis in 30 seconds..
Started task 226: TPSCountTimer, Sync.
Started task 227: NetworkPageRefreshTask, Async.
Started task 228: BootAnalysisTask, Async.
Started task 229: PeriodicAnalysisTask, Async.
Registered a new datasource: AdvancedAntiCheat
Registered a new datasource: ASkyBlock
Registered a new datasource: Essentials
Registered a new datasource: Factions
Registered a new datasource: Economy (Essentials Economy)
Verbose debug messages are enabled.
|                  | Enable (Feb 13, 07:32:27) ------------
| Feb 13, 07:32:27 | 12 ms     Initializing locale
| Feb 13, 07:32:27 | 155 ms    Init Database
| Feb 13, 07:32:27 | 62 ms     WebServer Initialization
| Feb 13, 07:32:27 | 11 ms     Register Listeners
| Feb 13, 07:32:27 | 1 ms      Task Registration
| Feb 13, 07:32:27 | Enabling bStats Metrics.
| Feb 13, 07:32:27 | 1203 ms   Enable
| ---------------- | --------------------------------------
Player Analytics Enabled.
NetworkSettings: Fetch Config settings from database..
NetworkSettings: No Config settings in database.
Cleaning the database.
Clean complete.
Ended task 217: DB Clean Task, Async.
Analysis | Starting Boot Analysis..
Analysis | Beginning analysis of user data..
Analysis | Fetching Data..
Analysis | Data Fetched (1 users, took 217ms), beginning Analysis of data..
Analysis | Analyzing additional data sources (3rd party)
|                  | Analysis (Feb 13, 07:33:03) ------------
| Feb 13, 07:33:03 | Analysis Fetch Phase
| Feb 13, 07:33:03 | started   Analysis: Fetch Phase
| Feb 13, 07:33:04 | 217 ms    Analysis: Fetch Phase
| Feb 13, 07:33:04 | started   Analysis: Data Analysis
| Feb 13, 07:33:04 | Analysis Phase
| Feb 13, 07:33:04 | 403 ms    Analysis: Data Analysis
| Feb 13, 07:33:04 | Analyzing additional data sources (3rd party)
| Feb 13, 07:33:04 | started   Analysis: 3rd party
| Feb 13, 07:33:04 | Additional Sources: 5
| Feb 13, 07:33:04 | started   Analysis: Source ASkyBlock
| Feb 13, 07:33:04 | started   Analysis: Source AdvancedAntiCheat
| Feb 13, 07:33:04 | started   Analysis: Source Economy (Essentials Economy)
| Feb 13, 07:33:04 | started   Analysis: Source Factions
| Feb 13, 07:33:04 | started   Analysis: Source Essentials
| Feb 13, 07:33:04 | 3 ms      Analysis: Source ASkyBlock
| Feb 13, 07:33:04 | 5 ms      Analysis: Source AdvancedAntiCheat
| Feb 13, 07:33:04 | 9 ms      Analysis: Source Economy (Essentials Economy)
| Feb 13, 07:33:04 | 10 ms     Analysis: Source Essentials
| Feb 13, 07:33:04 | 13 ms     Analysis: Source Factions
| Feb 13, 07:33:04 | 16 ms     Analysis: 3rd party
| Feb 13, 07:33:04 | 644 ms    Analysis: Total
| ---------------- | ----------------------------------------
Analysis | Analysis Complete. (took 644 ms) 
Ended task 228: BootAnalysisTask, Async.
First run of TPSCountTimer Task.
Analysis | Beginning analysis of user data..
Analysis | Fetching Data..
Analysis | Data Fetched (1 users, took 194ms), beginning Analysis of data..
Analysis | Analyzing additional data sources (3rd party)
|                  | Analysis (Feb 13, 07:37:38) ------------
| Feb 13, 07:37:38 | Analysis Fetch Phase
| Feb 13, 07:37:38 | started   Analysis: Fetch Phase
| Feb 13, 07:37:39 | 194 ms    Analysis: Fetch Phase
| Feb 13, 07:37:39 | started   Analysis: Data Analysis
| Feb 13, 07:37:39 | Analysis Phase
| Feb 13, 07:37:39 | 246 ms    Analysis: Data Analysis
| Feb 13, 07:37:39 | Analyzing additional data sources (3rd party)
| Feb 13, 07:37:39 | started   Analysis: 3rd party
| Feb 13, 07:37:39 | Additional Sources: 5
| Feb 13, 07:37:39 | started   Analysis: Source Essentials
| Feb 13, 07:37:39 | started   Analysis: Source AdvancedAntiCheat
| Feb 13, 07:37:39 | started   Analysis: Source ASkyBlock
| Feb 13, 07:37:39 | started   Analysis: Source Economy (Essentials Economy)
| Feb 13, 07:37:39 | started   Analysis: Source Factions
| Feb 13, 07:37:39 | 0 ms      Analysis: Source ASkyBlock
| Feb 13, 07:37:39 | 0 ms      Analysis: Source Essentials
| Feb 13, 07:37:39 | 1 ms      Analysis: Source AdvancedAntiCheat
| Feb 13, 07:37:39 | 1 ms      Analysis: Source Economy (Essentials Economy)
| Feb 13, 07:37:39 | 3 ms      Analysis: Source Factions
| Feb 13, 07:37:39 | 4 ms      Analysis: 3rd party
| Feb 13, 07:37:39 | 446 ms    Analysis: Total
| ---------------- | ----------------------------------------
Analysis | Analysis Complete. (took 446 ms) 
Ended task 229: PeriodicAnalysisTask, Async.
```

**Comment by SlimeDog** [comment]
In other words, data analysis for the plugins is fine. It just doesn't generate automatically.

**Comment by Rsl1122** [comment]
Here is the answer
`Ended task 229: PeriodicAnalysisTask, Async.`
Looks like there may be an extra cancel somewhere

**Comment by SlimeDog** [comment]
That would be a good find. It doesn't appear to be in 4.1.5-b3 or previous.

**Comment by SlimeDog** [comment]
It looks like it runs the boot analysis, after restart, and then one more, after which the PeriodicAnalysisTask is gone.
```
[09:35:13 INFO]: [Plan] Analysis | Starting Boot Analysis..
[09:35:13 INFO]: [Plan] Analysis | Beginning analysis of user data..
[09:35:13 INFO]: [Plan] Analysis | Fetching Data..
[09:35:14 INFO]: [Plan] Analysis | Data Fetched (1 users, took 221ms), beginning Analysis of data..
[09:35:14 INFO]: [Plan] Analysis | Analyzing additional data sources (3rd party)
[09:35:14 INFO]: [Plan] Analysis | Analysis Complete. (took 655 ms)
>plan status
[09:35:58 INFO]: » Plan Status
[09:35:58 INFO]:  • Tasks running: 12
[09:35:58 INFO]:  • Processes: 
[09:35:58 INFO]:    : | Feb 13, 09:35:23 | started   Request:GET /js/demo.js
[09:35:58 INFO]:  • Tasks: 
[09:35:58 INFO]:    Task 216: DBConnectionPingTask SQLite, Async.
[09:35:58 INFO]:    Task 218: ProcessQueueConsumer, Async.
[09:35:58 INFO]:    Task 219: ProcessQueueConsumer, Async.
[09:35:58 INFO]:    Task 220: ProcessQueueConsumer, Async.
[09:35:58 INFO]:    Task 221: ProcessQueueConsumer, Async.
[09:35:58 INFO]:    Task 222: ProcessQueueConsumer, Async.
[09:35:58 INFO]:    Task 223: ProcessQueueConsumer, Async.
[09:35:58 INFO]:    Task 224: ProcessQueueConsumer, Async.
[09:35:58 INFO]:    Task 225: ProcessQueueConsumer, Async.
[09:35:58 INFO]:    Task 226: TPSCountTimer, Sync.
[09:35:58 INFO]:    Task 227: NetworkPageRefreshTask, Async.
[09:35:58 INFO]:    Task 229: PeriodicAnalysisTask, Async.
[09:35:58 INFO]: »
>plan status
[09:36:50 INFO]: » Plan Status
[09:36:50 INFO]:  • Tasks running: 12
[09:36:50 INFO]:  • Processes: 
[09:36:50 INFO]:    : | Feb 13, 09:35:23 | started   Request:GET /js/demo.js
[09:36:50 INFO]:  • Tasks: 
[09:36:50 INFO]:    Task 216: DBConnectionPingTask SQLite, Async.
[09:36:50 INFO]:    Task 218: ProcessQueueConsumer, Async.
[09:36:50 INFO]:    Task 219: ProcessQueueConsumer, Async.
[09:36:50 INFO]:    Task 220: ProcessQueueConsumer, Async.
[09:36:50 INFO]:    Task 221: ProcessQueueConsumer, Async.
[09:36:50 INFO]:    Task 222: ProcessQueueConsumer, Async.
[09:36:50 INFO]:    Task 223: ProcessQueueConsumer, Async.
[09:36:50 INFO]:    Task 224: ProcessQueueConsumer, Async.
[09:36:50 INFO]:    Task 225: ProcessQueueConsumer, Async.
[09:36:50 INFO]:    Task 226: TPSCountTimer, Sync.
[09:36:50 INFO]:    Task 227: NetworkPageRefreshTask, Async.
[09:36:50 INFO]:    Task 229: PeriodicAnalysisTask, Async.
[09:36:50 INFO]: »
[09:39:48 INFO]: [Plan] Analysis | Beginning analysis of user data..
[09:39:48 INFO]: [Plan] Analysis | Fetching Data..
[09:39:48 INFO]: [Plan] Analysis | Data Fetched (1 users, took 198ms), beginning Analysis of data..
[09:39:49 INFO]: [Plan] Analysis | Analyzing additional data sources (3rd party)
[09:39:49 INFO]: [Plan] Analysis | Analysis Complete. (took 460 ms)
>plan status
[09:41:01 INFO]: » Plan Status
[09:41:01 INFO]:  • Tasks running: 11
[09:41:01 INFO]:  • Processes: 
[09:41:01 INFO]:    : | Feb 13, 09:40:43 | started   Request:GET /js/demo.js
[09:41:01 INFO]:  • Tasks: 
[09:41:01 INFO]:    Task 216: DBConnectionPingTask SQLite, Async.
[09:41:01 INFO]:    Task 218: ProcessQueueConsumer, Async.
[09:41:01 INFO]:    Task 219: ProcessQueueConsumer, Async.
[09:41:01 INFO]:    Task 220: ProcessQueueConsumer, Async.
[09:41:01 INFO]:    Task 221: ProcessQueueConsumer, Async.
[09:41:01 INFO]:    Task 222: ProcessQueueConsumer, Async.
[09:41:01 INFO]:    Task 223: ProcessQueueConsumer, Async.
[09:41:01 INFO]:    Task 224: ProcessQueueConsumer, Async.
[09:41:01 INFO]:    Task 225: ProcessQueueConsumer, Async.
[09:41:01 INFO]:    Task 226: TPSCountTimer, Sync.
[09:41:01 INFO]:    Task 227: NetworkPageRefreshTask, Async.
```

**Comment by SlimeDog** [comment]
With all plugins configured as _enabled:false_, the same outcome. Analysis runs once after boot analysis, then _PeriodicAnalysisTask_ disappears.

**Comment by Rsl1122** [comment]
https://github.com/Rsl1122/Plan-PlayerAnalytics/blob/master/Plan/src/main/java/com/djrapitops/plan/system/tasks/bukkit/PeriodicAnalysisTask.java#L31

Looks like it got in while refactoring anonymous tasks into their own classes. 

So this has been solved, just needs a fix.

**Comment by SlimeDog** [comment]
Good find. Am I the only one experiencing this, or has no one else noticed?

**Comment by SlimeDog** [comment]
Confirmed fixed in 4.1.6. Thank you.
